daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionCoderSpec.scala

```diff
@@ -340,6 +340,161 @@ class TransactionCoderSpec
       }
     }
 
+    def changeVersion(
+        value: com.daml.lf.value.ValueOuterClass.VersionedValue,
+        version: String,
+    ) =
+      value.toBuilder.setVersion(version).build()
+
+    "fail if try to encode a create node containing value with version different from node" in {
```

Comment:
This sounds like we fail during encoding but that seems to work fine. What fails is decoding. Should we fail to encode such values already?

Reply:
In Scala land the Value in GenNode are not wrapped in VersionedValue any more, so we cannot generate "such" value.

Reply:
ah I see, makes sense.